lib/redis/rack/connection.rb

```diff
@@ -35,7 +35,7 @@ def pool_options
         {
           size: @options[:pool_size],
           timeout: @options[:pool_timeout]
-        }
+        }.compact
```

Comment:
hey @shepmaster, sorry about this I was definitely wrong about being able to use `Hash#compact` on older Ruby versions. it failed the build on Ruby 2.0-2.3, so I think we might need to return to your other strategy of iterating over the keys and only selecting the ones that have values...

```ruby
         {
           size: @options[:pool_size],
           size: @options[:pool_size],
           timeout: @options[:pool_timeout]
           timeout: @options[:pool_timeout]
         }.reject { |key, value| value.nil? }.to_h
```

Reply:
> your other strategy of iterating over the keys and only selecting the ones that have values

My previous revision used your original technique of simple `if`s:

https://github.com/redis-store/redis-rack/blob/8b2374b07c7950afc6691a4f65c88f011ed672e7/lib/rack/session/redis.rb#L22-L24

Reply:
Updated.